File: app/src/main/java/com/example/neon/deardiary/notify/NotifyPresenter.java
package com.example.neon.deardiary.notify;

import com.example.neon.deardiary.data.Diary;
import com.example.neon.deardiary.data.source.DataSource;

import java.util.Calendar;


public class NotifyPresenter implements NotifyContract.Presenter {
    private DataSource mDataSource;
    private NotifyContract.View mView;

    NotifyPresenter(DataSource dataSource, NotifyContract.View view) {
        mDataSource = dataSource;
        mView = view;
        mView.setPresenter(this);
    }

    @Override
    public void start() {

    }

    @Override
    public Diary getDiary(Calendar c) {
        final Diary[] diary = new Diary[1];
        mDataSource.queryByDay(c, new DataSource.GetDiaryCallBack() {
            @Override
            public void onDiaryGot(Diary foundDiary) {
                diary[0] = foundDiary;
            }

            @Override
            public void onDataNotAvailable() {
                diary[0] = null;
            }
        });
        return diary[0];
    }

    @Override
    public void insertDiary(Diary diary) {
        mDataSource.insertDiary(diary);

    }
}

File: app/src/main/java/com/example/neon/deardiary/notify/NotifyPresenter.kt
package com.example.neon.deardiary.notify

import com.example.neon.deardiary.data.Diary
import com.example.neon.deardiary.data.source.DataSource

import java.util.Calendar


internal class NotifyPresenter(private val mDataSource: DataSource?, view: NotifyContract.View) : NotifyContract.Presenter {

    init {
        view.setPresenter(this)
    }

    override fun start() {

    }

    override fun getDiary(c: Calendar): Diary? {
        var foundDiary:Diary? = null
        mDataSource!!.queryByDay(c, object : DataSource.GetDiaryCallBack {
            override fun onDiaryGot(diary: Diary) {
                foundDiary = diary
            }

            override fun onDataNotAvailable() {
                foundDiary = null
            }
        })
        return foundDiary
    }

    override fun insertDiary(diary: Diary) {
        mDataSource!!.insertDiary(diary)

    }
}
